tools-local/unix/tool-helper.sh

```diff
@@ -2,15 +2,36 @@
 
 # Provides helper functions.
 
+# Checks if the specified path is valid.
+# If path is empty or does not exist then, displays a message stating the same, and exits with status 1. 
+exit_if_invalid_path()
+{
+    pathName="$1"
+    pathValue="$2"
+    usage="$3"
+
+    exit_if_arg_empty "$pathName" "$pathValue" "$usage"
+
+    if [ ! -f "$pathValue" ]; then
+        if [ ! -d "$pathValue" ]; then
```

Comment:
It is unknown if the given path is a file or a directory.  So I think we should check both conditions in tandem and not clubbed as  `[ ! -f "$pathValue" ] || [ ! -d "$pathValue" ]`  or `[ ! -f "$pathValue" ] && [ ! -d "$pathValue" ]`